.github/workflows/backend-code-checks.yml

```diff
@@ -0,0 +1,27 @@
+name: Backend Code Checks
+
+on:
+  pull_request:
+    paths-ignore:
+      - '*.md'
+      - 'docs/**'
+      - 'latest.json'
+
+permissions:
+  contents: read
+
+jobs:
+  validate-configs:
+    name: Validate Backend Configs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.23.5'
+          cache: true
+      
+      - name: Validate Backend Specs
+        uses: ./.github/actions/validate-backend-specs
```

Comment:
I don't think having this as a separate action gains us anything. If we run this action without the previous "checkout" or "setup-go" steps, then it just fails. IMO we remove that action.yml and just put those 2 steps here.